### ISSUE TITLE
Add guideline about use of podman vs docker commands for managing containers 

### ISSUE THREAD

**Issue by roldanbob** [redhat-documentation/supplementary-style-guide#32]
The OpenShift docs have recently moved away from including `docker` commands, replacing them with `podman` commands. Users find it confusing to have two sets of commands without clear guidance about the advantage of using one set over the other. In light of recent efforts to de-emphasize _docker_ terminology, unless there is a specific reason to use `docker` commands, we should use `podman` commands.

**Comment by bergerhoffer** [comment]
@redhat-documentation/ccs-style-council We currently have a product-specific glossary section. Should we consider whether we'll want to add a product-specific style guidance section for things like this?

Or is there another way we can incorporate specific technical info like this into our style guide?

**Comment by adellape** [comment]
FWIW, the `docker` package is not available in RHEL8: https://access.redhat.com/solutions/3696691

In OpenShift docs, we've been preferring showing `podman` for simple client commands (push/pull/login/run) for a while.

**Comment by pwright** [comment]
I think this is a technical issue, not a style issue, consider the following numbers:
![image](https://user-images.githubusercontent.com/5154224/96694769-c3ffe080-1380-11eb-9280-e99aa2df2154.png)
![image](https://user-images.githubusercontent.com/5154224/96694891-eabe1700-1380-11eb-9077-cf31297b83b3.png)

While users with a complete RH stack should use podman, there are users who will use docker for specific purposes. There were already guidelines about Docker (not from CSS), a link to those guidelines would be preferable, IMO.

**Comment by adellape** [comment]
> While users with a complete RH stack should use podman, there are users who will use docker for specific purposes. There were already guidelines about Docker (not from CSS), a link to those guidelines would be preferable, IMO.

The one I'm aware of is an internal link (therefore likely not appropriate here), unless you're referring to something else?
 
> Although, I'd be happy with something like "When technically equivalent, use `podman` instead of `docker` commands."

+1 to this.

**Comment by pwright** [comment]
@adellape I was referring to the internal link, and agree that it's not appropriate here, but maybe something like https://developers.redhat.com/blog/2019/02/21/podman-and-buildah-for-docker-users/ (thanks @jbyrne-redhat) would be helpful for new writers who want background.

**Comment by inoxx03** [comment]
> Although, I'd be happy with something like "When technically equivalent, use podman instead of docker commands."

TL;DR: +1 This would be a good minimal guideline (see below for details). 

We should refer to docker as the default container management tool in product documentation. The Podman [community docs](https://podman.io/whatis.html) even mention that the command syntax for both tools are highly equivalent (to the point that you can alias `docker` to `podman` and carry on using the same commands).

You could even parameterize or conditionalize the commands with AsciiDoc attributes when single sourcing, allowing you to keep `docker` in the community docs and `podman` in product.

@pwright would it be acceptable to link to https://developers.redhat.com/blog/2019/02/21/podman-and-buildah-for-docker-users/  from our style guide as an additional resource?

**Comment by pwright** [comment]
@inoxx03 works for me

**Comment by stoobie** [comment]
I've read through all the comments, and I still don't understand why this belongs in the style guide. Why would someone looking for this type of info consider looking in the style guide? Is the suggestion to add it to the glossary? If so, under which product? RHEL? OpenShift?

**Comment by leswilliams44** [comment]
Hi, @roldanbob. Are there any updates for this issue?

**Comment by roldanbob** [comment]
@leswilliams44 No, no new points to add on my end. 

I'm fine with either offering minimal guidance with links to other resources, including the information as part of a glossary entry, or leaving it out of the style guide altogether. Although the style guide might not be the appropriate vehicle, it does seem like it would make sense to codify our recommended approach somewhere. Originally, I raised this issue because I wasn't sure whether we had a recommended practice, or if we did, where I might find guidance. Writers on the RHEL, OpenShift, Red Hat Integration, and possibly other teams are likely to encounter questions about recommended usage.

**Comment by leswilliams44** [comment]
@cbudz and @bergerhoffer, Is this issue something we can vote on at the next meeting?

**Comment by bergerhoffer** [comment]
We could definitely bring it up for discussion. We have product-specific glossary terms, but right now we don't have product-specific style/guidance/conventions. It's definitely something that we can consider adding, but I'm not sure that we would add just this entry alone without other guidance.

It might be something that we just have to table until we figure out if this is a direction that we're going to go or not.

**Comment by bergerhoffer** [comment]
Upon further discussion, closing this issue as this is more of a product strategy item and is outside the scope of the general style guidelines.
